app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1834,6 +1835,22 @@ class PluginApi {
     addGlimmerSearchSuggestion(value);
   }
 
+  /**
+   * Force a given menu panel (search-menu, user-menu, etc) to be displayed as dropdown if ANY of the passed `classNames` are included in the `classList` of a menu panel.
+   * This can be useful for plugins as the default behavior is to add a 'slide-in' behavior to a menu panel if you are viewing on a small screen. eg. mobile.
+   * Sometimes when we are rendering the menu panel in a non-standard way we don't want this behavior and want to force the menu panel to be displayed as a dropdown.
+   *
+   * The `classNames` param can be passed as a single string or an array of strings. This way you can disable the 'slide-in' behavior for multiple menu panels.
+   *
+   * ```
+   * api.forceSearchMenuResultsAsDropdown(["search-menu-panel", "custom-class-for-menu-panel"]);
+   * ```
+   *
+   */
+  forceDropdownAnimationForMenuPanels(classNames) {
```

Comment:
I'm having a bit of trouble with the `Animation` in the method name here. The description of this plugin API entry is "Force a given menu panel to be displayed as dropdown" makes me think we can drop the `Animation` in the name.